test/uk/gov/hmrc/channelpreferences/controllers/PreferenceControllerSpec.scala

```diff
@@ -83,21 +88,176 @@ class PreferenceControllerSpec extends PlaySpec with ScalaFutures with MockitoSu
+          Entity(passedBackEntityId, Some(resolvedEntity_saUtr), nino = None, itsa = Some(passedBackItsaId))))
+
+      when(mockEntityResolverConnector.update(any[Entity]())(any[HeaderCarrier]()))
+        .thenReturn(Future.successful(
+          Entity(passedBackEntityId, Some(resolvedEntity_saUtr), nino = None, itsa = Some(passedBackItsaId))))
+
+      val postData: JsValue = Json.parse(s"""{"entityId": "$passedBackEntityId", "itsaId": "$passedBackItsaId"}""")
+      val fakePostRequest = FakeRequest("POST", "", Headers("Content-Type" -> "application/json"), postData)
+      val response = controller.confirm().apply(fakePostRequest)
+      status(response) mustBe OK
+      contentAsString(response) mustBe """{"reason":"itsaId successfully linked to entityId"}"""
+    }
+
+    // Case 1.5
+    """return UNAUTHORIZED (401) when entityId already has a different itsaId linked to it in entity resolver""" in new TestSetup {
+      reset(mockEntityResolverConnector)
+      reset(mockAuthConnector)
+
+      when(
+        mockAuthConnector.authorise[Option[String]](any[Predicate](), any[Retrieval[Option[String]]]())(
+          any[HeaderCarrier](),
+          any[ExecutionContext]()))
+        .thenReturn(Future.successful(Some("authTokenSaUtr")))
+
+      val passedBackEntityId = "passedBackEntityId"
+      val passedBackItsaId = "passedBackItsaId"
+
+      val resolvedEntity_saUtr = "resolvedEntity_saUtr"
+      val resolvedEntity_itsaId = "different_than_passedBackItsaId"
+      when(mockEntityResolverConnector.resolveBy(anyString())(any[HeaderCarrier]()))
+        .thenReturn(
+          Future.successful(
+            Entity(
+              passedBackEntityId,
+              saUtr = Some(resolvedEntity_saUtr),
+              nino = None,
+              itsa = Some(resolvedEntity_itsaId))))
+
+      when(mockEntityResolverConnector.update(any[Entity]())(any[HeaderCarrier]()))
+        .thenReturn(Future.successful(
+          Entity(passedBackEntityId, Some(resolvedEntity_saUtr), nino = None, itsa = Some(passedBackItsaId))))
+
+      val postData: JsValue = Json.parse(s"""{"entityId": "$passedBackEntityId", "itsaId": "$passedBackItsaId"}""")
+      val fakePostRequest = FakeRequest("POST", "", Headers("Content-Type" -> "application/json"), postData)
+      val response = controller.confirm().apply(fakePostRequest)
+      status(response) mustBe UNAUTHORIZED
+      contentAsString(response) mustBe """{"reason":"entityId already has a different itsaId linked to it in entity resolver"}"""
+      verify(mockEntityResolverConnector, never()).update(any[Entity]())(any[HeaderCarrier]())
+    }
+
+    // Case 1.6
+    """return UNAUTHORIZED (401) when itsaId is already linked to a different entityId in entity resolver""" in new TestSetup {
+      pending
+      // WARNING: this scenario seems already included in the previous one
+    }
+
+    // Case 2.1
+    // It seems not relevant to this channel-preferences service
```

Comment:
Without reference to `Case x.x` I cannot tell, so this comment only conflates the matter. I would remove the comment here and other comments refering to things I do not have in scope.